Add a bit of logic to use in the App Component

elmejdki opened the pull request:
To be honest, I wasn't knowing what I was doing, because the requirements
are really vag, I couldn't understand clearly what the requirements are,
so now I'm waiting for the review to improve the code, please give as
much details as possible.

elmejdki committed:
Add a bit of logic to use in the App Component

To be honest here I wasn't knowing I was doing cause the requirements
are realy vag, I couldn't understand clearly what the requirements are,
so now I'm waiting for the review to improve the code, please give as
much details as possible.

elmejdki committed:
Add a new line in the operate file for linter error

MuriloRoque commented on src/components/App.js:
```diff
@@ -1,6 +1,10 @@
+import calculate from '../logic/calculate';
+
+// eslint-disable-next-line no-console
+console.log(calculate({ total: 12, next: 23, operation: 'X' }, 'X'));
```
- [x] Please remove this `console.log` before the next review. I know you were testing your function, but it's recommended that you don't leave it in your code, alright?

MuriloRoque commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,121 @@
+      return {
+        total: 0,
+        next: 0,
+        operation: 'AC',
```
- [x] There's no need to make `operation` equal to `AC`, since you won't be using it anyway, do you agree? This would be a "reset" button, to make everything turn to its initial state.

MuriloRoque commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,121 @@
+        total: operate(total, next, '-'),
+        next: 0,
+        operation: '-',
+      };
```
- [x] From what I told you to use `if/else` statements before, these would be only one type of case here when the user types an operation.

MuriloRoque commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,121 @@
+        total,
+        next: Number(`${total}1`),
+        operation,
+      };
```
- [x] Just as I told you above, all of these buttons are a part of one case when the user presses a number, so you could make only one conditional for these.

MuriloRoque commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,121 @@
+      return {
+        total: 0,
+        // eslint-disable-next-line no-eval
+        next: eval(`${next} ${operation} ${total}`),
```
- [x] Why are you using `eval` here? I believe you can put the string literal directly instead of using this function. Can you please try another way?

MuriloRoque commented on src/logic/operate.js:
```diff
@@ -0,0 +1,23 @@
+    case 'X':
+      return firstNumber.times(secondNumber).toNumber();
+    case '%':
+      return firstNumber.div(100).toNumber();
```
- [x] Take a look at the `mod` function available in the `Big.js` library. it would avoid the use of `div` here, and make the operation more direct.

MuriloRoque reviewed (changes requested):
## Status: Changes requested :red_circle: 

### Hello @elmejdki :wave: 

### You started well this milestone, I am sorry that the requirements confused you. I will bring this to the TSE's group and see what they could change to make it better. I will try to be as much descriptive as I can in the requirements. If you need help in a Zoom call, I would advise you to fill out a form [here](https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-) and talk to a Full-Time TSE, I am sure they will help you understand. If you have any questions about my review, please come talk to me on Slack! :smile: Kindly check my recommendations below:

- [x] Check my comments in your code and you will understand, but I believe it would be better if you use `if/else` conditionals here, so you can have more freedom in choosing the right comparisons between the values.

Kindly make the requested changes and submit for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)

elmejdki committed:
Adjust code based on code review

 - changed the use of switch to if/else statements
 - Added some arrays, like operations and nums to reduce complexity and
 resume code
 - Removed the unnecasary console.log call in the App Component
 - improved the logic of calculate method
 - change the call to .div(100) to muliples(Big(0.01))

Forison commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,100 @@
+  const operations = ['%', '÷', '+', '-', 'X'];
+  const nums = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+  if (buttonName === 'AC') {
```
**[OPTIONAL]**
It relatively faster to use `if/else if` statement in this case. In this case, all the condition would run, It is said that this effect insignificant when the code is small, but I just wanted to draw to your attention.

elmejdki replied:
can't use the else if statement cause it prompts an error on the linters:
![image](https://user-images.githubusercontent.com/34653764/102550163-a4a1dd00-40bd-11eb-9866-9dae9a7e82e0.png)

Forison reviewed (approved):
# Status::Approved✔️
## Hi @elmejdki 
### Good job on addressing all the issues raised in the previous review🤗.
#### Check my comments below👇.
### Happy coding🎅
Addo Forison
